### DISCUSSION TITLE
[Feature request] Add Enum literals to autocomplete when type-appropriate

### DISCUSSION THREAD

**Discussion by nuiva** [microsoft/pylance-release#2583] [Enhancement]
I'd like the autocomplete to infer from types that I'm about to write an Enum value and immediately suggest the possible literals in the Enum class. Consider the following example:

![autocomplete_demo_image](https://user-images.githubusercontent.com/34218315/123092197-e62b9000-d432-11eb-9d44-8babfb781f0c.png)

In this example, I want the autocomplete to include `ENUM_VAL_A` and `ENUM_VAL_B` (which expand to `TestEnum.ENUM_VAL_A`) and I think these literals should be at the top of the list. The current behaviour is that I must type `TestEnum` (which isn't even guaranteed to be at the top of the suggestion list) before `ENUM_VAL_A` and B are included in the autocomplete suggestions.

Similarly, if I define
```python
def f(x: TestEnum) -> None:
    pass
```
then the literals should be suggested for the parameters of `f`.

**Comment by jakebailey** [comment]
Marking as a feature request; this would be a "deep completion", as the thing you'd need to suggest is the full `TestEnum.ENUM_VAL_A` expression (and we only go one level down).

Or, the first type-aware completion where we suggest things first if they would type check.

**Comment by kloringne** [comment]
oh so this is what youre so busy with... traitor!

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
